Objects/genericaliasobject.c

```diff
@@ -487,11 +487,49 @@ ga_reduce(PyObject *self, PyObject *Py_UNUSED(ignored))
                          alias->origin, alias->args);
 }
 
+static PyObject *
+ga_dir(PyObject *self, PyObject *Py_UNUSED(ignored))
+{
+    gaobject *alias = (gaobject *)self;
+    PyObject *dir = PyObject_Dir(alias->origin);
+    if (dir == NULL) {
+        return NULL;
+    }
+
+    int contains = 0;
+    PyObject *dir_entry = NULL;
+    for (const char * const *p = attr_exceptions; ; p++) {
+        if (*p == NULL) {
+            break;
+        }
+        else {
+            PyObject *dir_entry = PyUnicode_FromString(*p);
```

Comment:
This shadows the `dir_entry` on line 500! The `XDECREF` on line 523 will never do anything and you will leak this object upon errors.

Reply:
I've removed the type redefinition, also used `Py_CLEAR`. If I'm not missing anything, should be OK now.